renku/cli/_providers/dataverse.py

```diff
@@ -0,0 +1,254 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2019 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Dataverse API integration."""
+import pathlib
+import urllib
+import urllib.parse as urlparse
+
+import attr
+import requests
+
+from renku.cli._providers.api import ExporterApi, ProviderApi
+from renku.cli._providers.doi import DOIProvider
+from renku.models.datasets import Dataset, DatasetFile
+from renku.utils.doi import extract_doi
+
+DATAVERSE_API_PATH = 'api'
+
+DATAVERSE_VERSION_API = 'info/version'
+DATAVERSE_METADATA_API = 'datasets/export'
+DATAVERSE_FILE_API = 'access/datafile/:persistentId/'
+DATAVERSE_EXPORTER = 'schema.org'
+
+
+def check_dataverse_uri(url):
```

Comment:
this should probably be refactored to be a generic classmethod of the `ProviderApi` to simplify the factory function logic - but maybe in a separate PR.